pkg/config/expand.go

```diff
@@ -186,6 +187,33 @@ func (bp Blueprint) expandBackend(grp *Group) {
 	}
 }
 
+func kubectlProviderRequiredModule(grp *Group) (bool, Module) {
+	for _, mod := range grp.Modules {
+		if strings.Contains(mod.Source, "gke-cluster") || strings.Contains(mod.Source, "pre-existing-gke-cluster") {
+			return true, mod
+		}
+	}
+	return false, Module{}
+}
+
+func getModuleKubectlProviders(mod Module) map[string]TerraformProvider {
+	kubectlConf := Dict{}
```

Comment:
Some questions:
- Are we considering multiple gke-cluster, pre-existing-gke-cluster blocks in a single deployment group? It's rare but possible. (I was thinking probably we'll have an "alias" setting in the provider block representing the targeted cluster. It also gives us the ability to have multiple blocks of a same provider.)
- Does this work in a fresh deployment that no gke-cluster is provisioned and in consequence we don't have cluster info/credentials while creating the deployment? 
- Have we tried a scenario that we have a network, a multivpc, only a gke-cluster (a3 that applies some manifest inside) and nothing else; deploy the blueprint then remove the gke-cluster?

Reply:
> Some questions:
> 
> * Are we considering multiple gke-cluster, pre-existing-gke-cluster blocks in a single deployment group? It's rare but possible. (I was thinking probably we'll have an "alias" setting in the provider block representing the targeted cluster. It also gives us the ability to have multiple blocks of a same provider.)
> * Does this work in a fresh deployment that no gke-cluster is provisioned and in consequence we don't have cluster info/credentials while creating the deployment?
> * Have we tried a scenario that we have a network, a multivpc, only a gke-cluster (a3 that applies some manifest inside) and nothing else; deploy the blueprint then remove the gke-cluster?

Please find response in line.
- This change don't consider multiple gke-cluster, pre-existing-gke-cluster blocks in single deployment group, for simplicity multi deployment group could help as it aligns well with how current cluster-toolkit been setup.
- In this scenario, we expect to usage of pre-existing-gke-cluster module, in order to build the required cluster info/credentials while creating the deployment.
- Tried this scenario, was not able to remove the gke-cluster, as the kubectl provider depends on the gke-cluster module output. The gke-cluster module has an [internally defined](https://github.com/GoogleCloudPlatform/cluster-toolkit/blob/main/modules/scheduler/gke-cluster/main.tf#L345-L368) kubectl-apply module, so not sure if this scenario is valid in current state. Meanwhile tried gcluster destroy and it worked fine.

Reply:
- Okay then, thanks. Probably we need to reflect in the doc that we have this single gke-cluster limit per deployment group.

- The below was @nick-stroud 's concern:
> Tried this scenario, was not able to remove the gke-cluster, as the kubectl provider depends on the gke-cluster module output. The gke-cluster module has an [internally defined](https://github.com/GoogleCloudPlatform/cluster-toolkit/blob/main/modules/scheduler/gke-cluster/main.tf#L345-L368) kubectl-apply module, so not sure if this scenario is valid in current state. Meanwhile tried gcluster destroy and it worked fine.